src/rgw/rgw_rest_iam.cc

```diff
@@ -0,0 +1,147 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include <boost/tokenizer.hpp>
+
+#include "rgw_rest.h"
+#include "rgw_rest_iam.h"
+
+#include "rgw_request.h"
+#include "rgw_process.h"
+
+#include "rgw_rest_role.h"
+#include "rgw_rest_user_policy.h"
+
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_rgw
+
+void RGWHandler_REST_IAM::rgw_iam_parse_input()
+{
+  if (post_body.size() > 0) {
+    ldout(s->cct, 10) << "Content of POST: " << post_body << dendl;
+
+    if (post_body.find("Action") != string::npos) {
+      boost::char_separator<char> sep("&");
+      boost::tokenizer<boost::char_separator<char>> tokens(post_body, sep);
```

Comment:
iirc boost::tokenizer isn't thought to be very efficient--maybe not a concern for now

Reply:
@adamemerson thoughts?

Reply:
@mattbenjamin @adamemerson @cbodley : I have the exact same method for parsing sts_input also, with a few differences. I could have made this into a common method that parses output for both. But later when we decide to have STS, IAM served at a different hostname:portname, then these methods will already be separate. Is this ok for now?